packages/angular_devkit/build_facade/src/architect_spec.ts

```diff
@@ -0,0 +1,248 @@
+/**
+ * @license
+ * Copyright Google Inc. All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import { normalize } from '@angular-devkit/core';
+import { ErrorObservable } from 'rxjs/observable/ErrorObservable';
+import { concatMap, tap, toArray } from 'rxjs/operators';
+import {
+  Architect,
+  ConfigurationNotFoundException,
+  InvalidBuilderException,
+  InvalidRunnerException,
+  ProjectNotFoundException,
+  TargetNotFoundException,
+  TargetNotSupportedException,
+  TargetOptions,
+} from './architect';
+import { Workspace } from './workspace';
+
+
+describe('Architect', () => {
+  const props = {
+    name: 'spec',
+    version: 1,
+    root: 'src',
+    buildSystem: './test/test-build-system',
+    projects: {
+      app: {
+        root: 'src/app',
+        projectType: 'application',
+        default: true,
+        targets: {
+          browser: {
+            targetType: 'browser',
+            default: true,
+            browserOption: 1,
+          },
+          test: {
+            targetType: 'karma',
+            karmaOption: 2,
+          },
+          lint: {
+            targetType: 'lint',
+          },
+        },
+        configurations: {
+          prod: {
+            targetType: 'browser',
+            optimize: true,
+          },
+        },
+      },
+      e2e: {
+        root: 'src/lib',
+        projectType: 'library',
+        buildSystem: 'another-build-system',
+        targets: {
+          browser: {
+            targetType: 'build-lib',
+            default: true,
+            libOption: 3,
+          },
+        },
+      },
+    },
+  };
+
+  it('works', (done) => {
+    const workspace = new Workspace(normalize('/path/to/workspace'), props);
+    workspace.load().subscribe({
+      complete: () => {
+        const architect = new Architect(workspace);
+        const targetOptions = architect.createTargetOptionsFromWorkspace();
```

Comment:
To me this feels like two different things;

1. workspace is just an interface, loaded from the `.angular.json`. There could be an `Architect.loadWorkspaceFromHost(host: Host, path: Path)` and `Architect.validateWorkspace(workspace: JsonObject): Workspace`.
2. Architect could be workspace agnostic. So the workspace isn't passed in the constructor, but to each methods.
3. Architect could take either a workspace and a list of names, or just a list of JSON objects to validate and build.

Reply:
I don't quite get what you mean here. Can we talk about this?